DT-432 Migrate Gnome Tetraves to Core22/Gnome42

sergio-costas opened the pull request:
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please check only the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:

* OS (please include version):
* Any other relevant environment information:

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

sergio-costas committed:
DT-432 Migrate Gnome Tetraves to Core22/Gnome42

sergio-costas committed:
Added parse-info

sergio-costas committed:
Removed unneeded dependency

sergio-costas committed:
Removed 'craftctl set version'

'craftctl set version' is not needed when a parse-info statement
is available.

sergio-costas commented:
@hellsworth Removed 'craftctl set version'

hellsworth commented on patches/patch1.diff:
```diff
@@ -0,0 +1,10026 @@
+diff --git a/data/meson.build b/data/meson.build
```
Please add a comment section at the top of the patch saying what it is for.

sergio-costas replied:
Done.

hellsworth commented on snapcraft.yaml:
```diff
@@ -23,25 +23,25 @@ slots:
-      snapcraftctl pull
-      snapcraftctl set-version $(git describe --tags --abbrev=10)
+      craftctl default
+      patch -p1 < $CRAFT_PROJECT_DIR/patches/patch1.diff
```
Consider using *.diff here in case there are other patches. Also please rename the patch to something more intuitive like fix-translations.diff or something.

sergio-costas replied:
Confirmed: bash wildcards expansion guarantees alphabetical order.

sergio-costas replied:
But using *.diff doesn't work, bash complains. A for loop must be used.

hellsworth reviewed (changes requested):
A couple of changes requested to help us remember what the patch is.

sergio-costas committed:
Separated patches

This patch separates all the patches in several files, and
adds comments about what do each one.

hellsworth reviewed (approved):
Looks great thanks!
